In [14]:
import numpy as np
import pandas as pd
import plotly.express as px


In [3]:
log = pd.read_csv("./data_files/logs.csv")

/var/folders/t9/_3gzl1xs2hg3nx4hkm8g1r200000gn/T/ipykernel_31337/2326660139.py:1: DtypeWarning: Columns (34,38,39,40,47,50,53,54,57,58,59,60,62,68,70,72,73,74,76,77,78,81,85,88,89,90,91,92,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,118,120,121,122,123,129,131) have mixed types. Specify dtype option on import or set low_memory=False.
  log = pd.read_csv("./data_files/logs.csv")


In [4]:
log

row_id  player_id  school                     wave     session  \
0              1    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
1              2    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
2              3    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
3              4    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
4              5    6427001    3561  3561 (T,TH) Spring 2013   Session 1   
...          ...        ...     ...                      ...         ...   
2106592  2106593   65670032    1531              Single wave  Session 10   
2106593  2106594   65670032    1531              Single wave  Session 10   
2106594  2106595   65670032    1531              Single wave  Session 10   
2106595  2106596   65670032    1531              Single wave  Session 10   
2106596  2106597   65670032    1531              Single wave  Session 10   

               date  event_id                  event_description  \
0        2013-03-07       600                 Player selects age   
1        2013-03-07       601              Player selects gender   
2        2013-03-07       602   Player commits to age and gender   
3        2013-03-07       603              Player selects avatar   
4        2013-03-07       604  Player commits to avatar and name   
...             ...       ...                                ...   
2106592  2013-12-13       706               Player pans epilogue   
2106593  2013-12-13       705        Player closes Epilogue Item   
2106594  2013-12-13         1                Player ends session   
2106595  2013-12-13       706               Player pans epilogue   
2106596  2013-12-13       708            Player hits done button   

          event_category event_time  ...  piece2_id  piece3_id  \
0        Avatar Creation     3M 18S  ...        NaN        NaN   
1        Avatar Creation     3M 19S  ...        NaN        NaN   
2        Avatar Creation     3M 25S  ...        NaN        NaN   
3        Avatar Creation     3M 30S  ...        NaN        NaN   
4        Avatar Creation     4M 12S  ...        NaN        NaN   
...                  ...        ...  ...        ...        ...   
2106592         Epilogue  1H 36M 1S  ...        NaN        NaN   
2106593         Epilogue  1H 36M 5S  ...        NaN        NaN   
2106594          General  1H 36M 7S  ...        NaN        NaN   
2106595         Epilogue  1H 36M 7S  ...        NaN        NaN   
2106596         Epilogue  1H 36M 7S  ...        NaN        NaN   

        opponent_point_loss player_points opponent_points  old_rating  \
0                       NaN           NaN             NaN         NaN   
1                       NaN           NaN             NaN         NaN   
2                       NaN           NaN             NaN         NaN   
3                       NaN           NaN             NaN         NaN   
4                       NaN           NaN             NaN         NaN   
...                     ...           ...             ...         ...   
2106592                 NaN           NaN             NaN         NaN   
2106593                 NaN           NaN             NaN         NaN   
2106594                 NaN           NaN             NaN         NaN   
2106595                 NaN           NaN             NaN         NaN   
2106596                 NaN           NaN             NaN         NaN   

         new_rating  priority_type_that_went_to_zero  epilogue_item_id  \
0               NaN                              NaN               NaN   
1               NaN                              NaN               NaN   
2               NaN                              NaN               NaN   
3               NaN                              NaN               NaN   
4               NaN                              NaN               NaN   
...             ...                              ...               ...   
2106592         NaN                              NaN               NaN   
2106593         Na

In [5]:
player1_log = pd.read_csv("./data_files/player-6427031 copy.csv")
player1_log

/var/folders/t9/_3gzl1xs2hg3nx4hkm8g1r200000gn/T/ipykernel_31337/1295355672.py:1: DtypeWarning: Columns (38,39,40,47,88,89,90,91,92,129,131) have mixed types. Specify dtype option on import or set low_memory=False.
  player1_log = pd.read_csv("./data_files/player-6427031 copy.csv")


row_id  player_id  school                         wave    session  \
0     308289    6427031    3561  3561 YES! Program Fall 2013  Session 1   
1     308290    6427031    3561  3561 YES! Program Fall 2013  Session 1   
2     308291    6427031    3561  3561 YES! Program Fall 2013  Session 1   
3     308292    6427031    3561  3561 YES! Program Fall 2013  Session 1   
4     308293    6427031    3561  3561 YES! Program Fall 2013  Session 1   
...      ...        ...     ...                          ...        ...   
7326  315615    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7327  315616    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7328  315617    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7329  315618    6427031    3561  3561 YES! Program Fall 2013  Session 8   
7330  315619    6427031    3561  3561 YES! Program Fall 2013  Session 8   

            date  event_id                                  event_description  \
0     2013-10-23       600                                 Player selects age   
1     2013-10-23       601                              Player selects gender   
2     2013-10-23       602                   Player commits to age and gender   
3     2013-10-23       603                              Player selects avatar   
4     2013-10-23       604                  Player commits to avatar and name   
...          ...       ...                                                ...   
7326  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7327  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7328  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7329  2013-12-11       815  Player cycles through Opportunity Cards using ...   
7330  2013-12-11       816                                  Player picks card   

       event_category  event_time  ...  piece2_id  piece3_id  \
0     Avatar Creation         12S  ...        NaN        NaN   
1     Avatar Creation         14S  ...        NaN        NaN   
2     Avatar Creation         15S  ...        NaN        NaN   
3     Avatar Creation         18S  ...        NaN        NaN   
4     Avatar Creation         36S  ...        NaN        NaN   
...               ...         ...  ...        ...        ...   
7326   Priority Sense  54H 4M 54S  ...        NaN        NaN   
7327   Priority Sense  54H 4M 55S  ...        NaN        NaN   
7328   Priority Sense  54H 4M 56S  ...        NaN        NaN   
7329   Priority Sense  54H 4M 57S  ...        NaN        NaN   
7330   Priority Sense  54H 5M 53S  ...        NaN        NaN   

     opponent_point_loss player_points opponent_points  old_rating  \
0                    NaN           NaN             NaN         NaN   
1                    NaN           NaN             NaN         NaN   
2                    NaN           NaN             NaN         NaN   
3                    NaN           NaN             NaN         NaN   
4                    NaN           NaN             NaN         NaN   
...                  ...           ...             ...         ...   
7326                 NaN           NaN             NaN         NaN   
7327                 NaN           NaN             NaN         NaN   
7328                 NaN           NaN             NaN         NaN   
7329                 NaN           NaN             NaN         NaN   
7330                 NaN           NaN             NaN         NaN   

      new_rating  priority_type_that_went_to_zero  epilogue_item_id  \
0            NaN                              NaN               NaN   
1            NaN                              NaN               NaN   
2            NaN                              NaN               NaN   
3            NaN                              NaN               NaN   
4            NaN                              NaN               NaN   
...          ...                              ...               ...   
7326         NaN         

In [6]:
player_1_notNull = player1_log.iloc[:, 15:]
player_1_notNull[player_1_notNull.notna() == True & player_1_notNull.columns.str.contains("_id")].select_dtypes(exclude = 'float')


skill_id object_id object_locked object_first_time_unlocked  \
0         NaN       NaN           NaN                        NaN   
1         NaN       NaN           NaN                        NaN   
2         NaN       NaN           NaN                        NaN   
3         NaN       NaN           NaN                        NaN   
4         NaN       NaN           NaN                        NaN   
...       ...       ...           ...                        ...   
7326      NaN       NaN           NaN                        NaN   
7327      NaN       NaN           NaN                        NaN   
7328      NaN       NaN           NaN                        NaN   
7329      NaN       NaN           NaN                        NaN   
7330      NaN       NaN           NaN                        NaN   

     scan_start_point scan_end_point object_scan_result animatic_id  \
0                 NaN            NaN                NaN         NaN   
1                 NaN            NaN                NaN         NaN   
2                 NaN            NaN                NaN         NaN   
3                 NaN            NaN                NaN         NaN   
4                 NaN            NaN                NaN         NaN   
...               ...            ...                ...         ...   
7326              NaN            NaN                NaN         NaN   
7327              NaN            NaN                NaN         NaN   
7328              NaN            NaN                NaN         NaN   
7329              NaN            NaN                NaN         NaN   
7330              NaN            NaN                NaN         NaN   

     avatar_gender avatar_name  ... clip3_validity opponent_attack_id  \
0              NaN         NaN  ...            NaN                NaN   
1              NaN         NaN  ...            NaN                NaN   
2              NaN         NaN  ...            NaN                NaN   
3              NaN         NaN  ...            NaN                NaN   
4              NaN         NaN  ...            NaN                NaN   
...            ...         ...  ...            ...                ...   
7326           NaN         NaN  ...            NaN                NaN   
7327           NaN         NaN  ...            NaN                NaN   
7328           NaN         NaN  ...            NaN                NaN   
7329           NaN         NaN  ...            NaN                NaN   
7330           NaN         NaN  ...            NaN                NaN   

     chosen_opponent_attack_type_id strategy_id piece_id_2 piece1_id  \
0                               NaN         NaN        NaN       NaN   
1                               NaN         NaN        NaN       NaN   
2                               NaN         NaN        NaN       NaN   
3                               NaN         NaN        NaN       NaN   
4                               NaN         NaN        NaN       NaN   
...                             ...         ...        ...       ...   
7326                            NaN         NaN        NaN       NaN   
7327                            NaN         NaN        NaN       NaN   
7328                            NaN         NaN        NaN       NaN   
7329                            NaN         NaN        NaN       NaN   
7330                            NaN         NaN        NaN       NaN   

     piece2_id piece3_id priority_type_that_went_to_zero  \
0          NaN       NaN                             NaN   
1          NaN       NaN                             NaN   
2          NaN       NaN                             NaN   
3          NaN       NaN                             NaN   
4          NaN       NaN                             NaN   
...        ...       ...                             ...   
7326       NaN       NaN                             NaN   
7327       NaN       NaN                             NaN   
7328       NaN       NaN                             

In [7]:
player_1_notNull.dtypes

skill_level_know                   float64
skill_level_priority               float64
skill_level_people                 float64
skill_level_refusal                float64
skill_level_me                     float64
                                    ...   
old_rating                         float64
new_rating                         float64
priority_type_that_went_to_zero     object
epilogue_item_id                   float64
item_locked_unlocked_state          object
Length: 117, dtype: object

In [8]:
pd.options.display.max_rows = 4000
p = player1_log.iloc[:, 15:].notna().sum()
p.sort_values(ascending = False)

event_type                         7331
name_of_file                       7331
minigame_level                     3857
starter_statement_id                915
npc_id                              885
scan_start_point                    712
scan_end_point                      712
priority_type                       611
opportunity_card_id                 519
object_id                           497
card_id                             487
npc_fact_data_type                  480
pan_scene_start_left                426
pan_scene_end_left                  426
object_scan_result                  336
invitation_id                       297
fact_card_id                        289
sense_id                            233
opponent_its_on_statement_id        225
inviting_npc_id                     198
inviting_npc_rank_id                198
aa_level_id                         184
object_first_time_unlocked          161
object_locked                       161
chunk_id                            157


In [9]:
player1_log["event_id"].unique()

array([ 600,  601,  602,  603,  604,  605,  210,  211,  213,  212,  207,
        205,  221,  220,  300,  305,  302,  303,  304,  306,  203,  100,
        102,  200,  219,  216,  217,  218,  204, 1003, 1004,  900,  901,
        904, 1002,  902,  903,  905,  906,  907,  908,  909,  910,  911,
       1005, 1001,  800,  801,  814,  815,  816,  803,  804,  805,  806,
        807,  808,  809,  810,  818,  811,  400,  401,  402,  405,  406,
        411,  403,  404,  412,  413,  414,  415,  416,  417,  418,  419,
        407,  408,  420, 1000,  409,  410,  103,  104,  105,  703,  706,
        707,    1,  708,  500,  501,  502,  503,  504,  505,  506,  507,
        508,  509,  510,  511,  512,  513,  514,  515,  516,  517,  301,
        704,  705,  101,  817,  912,  812,  813])

## Analysing correct choices
| event_category      | event_id           | event_description  | validity |
| ------------- |:-------------:| :-----:| -----:|
| Knowledge Minigame      | 402 | Player selects "True" | true or false |
| Knowledge Minigame      | 403      |  Player selects "False" |  |

In [10]:
# dataframe for validity check on each action

validity_check = log[['player_id', 'date', 'event_id', 'event_category', 'event_description', 'validity']]
validity_check = validity_check[validity_check.validity.notnull()]
# remove ']' string from end of each boolean
validity_check['validity'] = validity_check['validity'].str.strip(']')

validity_check.to_csv('validity_check.csv')
validity_check

player_id        date  event_id      event_category  \
1288       6427001  2013-03-07       403  Knowledge Minigame   
1299       6427001  2013-03-07       408  Knowledge Minigame   
1303       6427001  2013-03-07       402  Knowledge Minigame   
1308       6427001  2013-03-07       404  Knowledge Minigame   
1330       6427001  2013-03-07       404  Knowledge Minigame   
...            ...         ...       ...                 ...   
2106125   65670032  2013-12-13       408  Knowledge Minigame   
2106148   65670032  2013-12-13       404  Knowledge Minigame   
2106170   65670032  2013-12-13       402  Knowledge Minigame   
2106175   65670032  2013-12-13       403  Knowledge Minigame   
2106195   65670032  2013-12-13       408  Knowledge Minigame   

                                         event_description validity  
1288                                Player selects "False"     true  
1299     Player selects "Use Fact" to use chosen fact card    false  
1303                                 Player selects "True"     true  
1308                              Player selects "Opinion"     true  
1330                              Player selects "Opinion"    false  
...                                                    ...      ...  
2106125  Player selects "Use Fact" to use chosen fact card     true  
2106148                           Player selects "Opinion"     true  
2106170                              Player selects "True"     true  
2106175                             Player selects "False"     true  
2106195  Player selects "Use Fact" to use chosen fact card     true  

[48128 rows x 6 columns]

In [11]:
survey = pd.read_csv("./data_files/S5_scores_cleaned.csv")
survey

player_id  weeks  S5_mean
0    6427001.0      0    3.900
1    6427001.0      3    3.900
2    6427001.0      6    3.875
3    6427027.0      0    3.800
4    6427027.0      3    4.000
5    6427027.0      6    4.000
6    6427027.0     12    3.800
7    6427027.0     24    3.600
8    6427030.0      0    2.900
9    6427030.0      3    3.000
10   6427030.0      6    3.200
11   6427030.0     12    3.300
12   6427030.0     24    3.500
13   6427032.0      0    4.000
14   6427032.0      3    3.900
15   6427032.0      6    4.000
16   6427032.0     12    3.900
17   6427032.0     24    3.800
18   6427034.0      0    4.000
19   6427034.0      3    3.800
20   6427034.0      6    4.000
21   6427034.0     12    3.900
22   6427034.0     24    4.000
23   6427036.0      0    3.800
24   6427036.0      3    3.800
25   6427036.0      6    3.700
26   6427036.0     12    4.000
27   6427037.0      0    4.000
28   6427037.0      6    4.000
29   6427037.0     12    4.000
30   6427038.0     12    3.100
31   6427039.0      0    4.000
32   6427039.0      3    4.000
33   6427039.0      6    4.000
34   6427039.0     12    4.000
35   6427039.0     24    4.000
36   6427040.0      0    3.900
37   6427040.0      3    3.800
38   6427040.0      6    4.000
39   6427040.0     12    3.900
40   6427041.0      0    3.900
41   6427041.0      3    3.800
42   6427041.0      6    3.000
43   6427041.0     12    4.000
44   6427041.0     24    3.200
45   6429001.0      0    3.800
46   6430002.0      0    4.000
47   6430002.0      3    4.000
48   6430002.0      6    4.000
49   6430002.0     12    4.000
50   6430002.0     24    4.000
51   6430003.0      0    4.000
52   6430003.0      3    4.000
53   6430003.0      6    3.700
54   6430004.0      0    3.900
55   6430004.0      3    3.700
56   6430004.0      6    3.900
57   6430004.0     12    3.800
58   6430008.0      0    4.000
59   6430008.0      3    3.700
60   6430008.0      6    3.700
61   6430008.0     12    4.000
62   6430008.0     24    3.900
63   6486001.0      0    4.000
64   6486001.0     24    3.700
65   6486002.0      0    4.000
66   6486002.0      3    4.000
67   6486002.0      6    3.800
68   6486002.0     12    4.000
69   6486002.0     24    3.900
70   6486003.0      0    4.000
71   6486003.0      3    4.000
72   6486003.0     12    4.000
73   6486003.0     24    4.000
74   6486004.0      0    3.900
75   6486004.0      3    3.900
76   6486004.0      6    4.000
77   6486004.0     12    3.900
78   6486005.0      0    3.500
79   6486005.0      3    3.500
80   6486005.0      6    3.600
81   6486005.0     12    3.700
82   6486005.0     24    4.000
83   6486009.0      0    4.000
84   6486009.0      3    3.100
85   6486009.0     12    3.800
86   6486011.0      0    4.000
87   6486011.0      3    4.000
88   6486011.0      6    4.000
89   6486011.0     12    4.000
90   6486013.0      0    4.000
91   6486018.0      0    1.300
92   6486018.0      3    3.700
93   6486018.0      6    3.100
94   6486019.0      0    3.000
95   6486019.0      3    4.000
96   6486019.0      6    3.900
97   6486019.0     12    4.000
98   6486020.0      0    4.000
99   6486023.0      0    4.000
100  6486023.0      3    3.900
101  6486023.0      6    4.000
102  6486023.0     12    4.000
103  6486025.0      0    3.900
104  6486025.0      6    4.000
105  6486025.0     12    4.000
106  6486026.0      0    4.000
107  6486026.0      3    4.000
108  6486026.0      6    4.000
109  6486026.0     12    4.000
110  6486029.0      0    4.000
111  6486029.0      3    4.000
112  6486029.0      6    4.000
113  6486029.0     12    3.500
114  6486032.0      0    4.000
115  6486032.0      3    4.000
116  6486032.0      6    4.000
117  6486032.0     12    3.900
118  6486033.0      0    4.000
119  6486033.0      3    4.000
120  6486033.0      6    4.000
121  6486033.0     12    4.000
122  6486034.0      0    3.900
123  6486038.0      0    4.000
124  6486038.0      3    4.000
125  6486038.0      6    4.000
126  6486038.0     12    4.000
127  6486039.0      0    4.000
128  6

### Plot Survey Score to Event time for each participant

In [12]:
# event time is given to us in seconds

#sample last row for each id, from each df

# get ingame timelogs for each participant
timelogs = log.groupby("player_id").last()
timelogs = timelogs[['event_time']]

# get final survey response
final_survey = survey.groupby("player_id").last()
final_survey = final_survey[['S5_mean']]
final_survey

S5_mean
player_id         
6427001.0    3.875
6427027.0    3.600
6427030.0    3.500
6427032.0    3.800
6427034.0    4.000
6427036.0    4.000
6427037.0    4.000
6427038.0    3.100
6427039.0    4.000
6427040.0    3.900
6427041.0    3.200
6429001.0    3.800
6430002.0    4.000
6430003.0    3.700
6430004.0    3.800
6430008.0    3.900
6486001.0    3.700
6486002.0    3.900
6486003.0    4.000
6486004.0    3.900
6486005.0    4.000
6486009.0    3.800
6486011.0    4.000
6486013.0    4.000
6486018.0    3.100
6486019.0    4.000
6486020.0    4.000
6486023.0    4.000
6486025.0    4.000
6486026.0    4.000
6486029.0    3.500
6486032.0    3.900
6486033.0    4.000
6486034.0    3.900
6486038.0    4.000
6486039.0    4.000
6486040.0    3.900
6506008.0    3.300
6506009.0    4.000
6546001.0    3.800
6546005.0    3.700
6546006.0    4.000
6546008.0    4.000
6546011.0    3.800
6546014.0    3.900
6566001.0    4.000
6566003.0    4.000
6567002.0    4.000
6567003.0    4.000
6567005.0    3.100
6606001.0    3.600
6606002.0    3.100
6606003.0    4.000
6606004.0    4.000
6606005.0    4.000
6606006.0    4.000
6606011.0    3.800
6607006.0    4.000
6607007.0    4.000
6607008.0    3.900
6607009.0    4.000
6607010.0    4.000
6608002.0    4.000

In [19]:
ingametimeSurvey = pd.merge(timelogs, final_survey, on='player_id')
ingametimeSurvey

event_time    61
S5_mean       61
dtype: int64

In [16]:
#plot
px.scatter(x=ingametimeSurvey.event_time, y=ingametimeSurvey.S5_mean)